### DISCUSSION TITLE
Daily electrical consumption a little more precisely in Modbus?

### DISCUSSION THREAD

**Discussion by maddinhome** [OStrama/weishaupt_modbus#52] [Ideas]
Can you read the daily electrical consumption a little more precisely in Modbus? The WEMPortal integration has more precise values. I would need the more precise values ​​for further integration... in the SMA portal to see the power consumption of the heat pump.

WEMPortal Integration:

![Screenshot 2024-11-28 112532WEM](https://github.com/user-attachments/assets/c018c03c-2db5-4ffa-af08-45b651da99b3)

Modbus:

![Screenshot 2024-11-28 112721](https://github.com/user-attachments/assets/74f4a19e-b2bd-4810-89ad-d03f3bef3503)

Then we could do without the WEMPortal. The connection to the WEMPortal does not always seem to be reliable. I would also prefer to stay in my own network and that is possible by reading out the Modbus. 

Thank you

**Comment by MadOne** [comment]
Hi,

The values provided by modbus are always floored to Integer. By now we cant provide more acurate values. The local Webinterface provides more accurate values. We are currently working on integrating the webif values.

**Comment by maddinhome** [comment]
local web interface? How can I achieve this... where did you get that? It's new to me that something like this even exists. ugh

`We are currently working on integrating the webif values.`

I'm very excited to see what else comes from you. If only I could do that too...

**Comment by MadOne** [comment]
You have to enable it at the heatpump in OEM mode.
Go to Fachmann. When prompted for an code enter 21. Be carefull, this allowes you to change values you are not supposed to do. 
In the options there you can enable json.

I think it has to be enabled in the user settings too. But im not sure about that.

**Reply by GrimmiMeloni** [reply]
Wasn't aware of this one. Nice.

> I think it has to be enabled in the user settings too. But im not sure about that.

It's not necessary. Activating in OEM suffices.

Bonus question: Have you explored if the JSON contains the current power consumption? The unit does not expose this via ModbusTCP, but I have seen it on the display's unit.

**Reply by MadOne** [reply]
In info -> Wärmepumpe i see:
Soll-Leistung & Ist-Leistung
But i have no clue if this is thermal or electricial power. 

Sadly this is no json, it is a webserver. So i have to crawl all that pages and extract values. 
Just fonished the Info menu.

Just go to the IP of your heatpump and create an account. Then you can see all.

**Reply by GrimmiMeloni** [reply]
> In info -> Wärmepumpe i see:
Soll-Leistung & Ist-Leistung

In this screen on the local webserver, or do you mean on the unit itself?

<img width="1161" alt="image" src="https://github.com/user-attachments/assets/179d5a5d-7c5d-4c64-a864-025820eedee1">

Asking, because in the above screen I cannot see said values.

**Reply by MadOne** [reply]
I mean the webserver. In the Right Menu below Warmwassertemperatur. Ich have 56 Values there in the menu you cut off.

**Reply by GrimmiMeloni** [reply]
Interesting. I only have about ~30 - and those energy values are missing. Do you run a different version of the CPU firmware (mine being 5.3 rev 9)?

**Comment by MadOne** [comment]
![image](https://github.com/user-attachments/assets/0c4ba75d-3b68-400a-aaaf-6abd8a886991)
![Bildschirmfoto vom 2024-12-06 19-23-36](https://github.com/user-attachments/assets/c5e7becb-34c9-4147-a860-4e5d135ea2e2)
![image](https://github.com/user-attachments/assets/e73aeaed-816b-4f63-bccf-07951c297b40)

**Comment by exp1338** [comment]
My CPU has FW 5.0. Now option to enable webinterface. :(
